File: Binary_Tree_Preorder_Traversal.java
class Binary_Tree_Preorder_Traversal {
    public List<Integer> preorderTraversal(TreeNode root) {
        List <Integer> list = new ArrayList<>();
        preOrder(root,list);
        return list;
    }
    private void preOrder(TreeNode root,List list){
        if(root==null)
        return;
        list.add(root.val);
        preOrder(root.left,list);
        preOrder(root.right,list);
    }
}

File: BinaryTreePreorderTraversal.kt
class BinaryTreePreorderTraversal {
    private val res = mutableListOf<Int>()
    fun preorderTraversal(root: TreeNode?): List<Int> {
        res.clear()
        preorder(root)
        return res
    }
    private fun preorder(node: TreeNode?) {
        if (node == null) return
        res.add(node.`val`)
        preorder(node.left)
        preorder(node.right)
    }
}

